File: 1/1-17.hs
double x = x * 2
halve x = x `div` 2

multiply a b
  | b == 0 = 0
  | even b = multiply (double a) (halve b)
  | otherwise = a + multiply a (b - 1)

main = do
  let a = 5
  let b = 7
  print $ "Multiplication of " ++ show a ++ " and " ++ show b ++ " is: " ++ show (multiply a b)

File: 1/1-17.ts
function double(x: number): number {
    return x * 2;
}

function halve(x: number): number {
    return x / 2;
}

function multiply(a: number, b: number): number {
    if (b === 0) {
        return 0;
    } else if (b % 2 === 0) {
        return multiply(double(a), halve(b));
    } else {
        return a + multiply(a, b - 1);
    }
}

const a = 5;
const b = 7;
console.log(`Multiplication of ${a} and ${b} is: ${multiply(a, b)}`);